packages/generator/src/artifacts/js/css-fn.ts

```diff
@@ -100,17 +100,24 @@ export function generateCssFn(ctx: Context) {
           utility.hasShorthand
             ? `(prop, value) => {
               const key = resolveShorthand(prop)
-              const propKey = classNames.get(key) || hypenateProperty(key)
+              const propKey = classNameByProp.get(key) || hypenateProperty(key)
               return { className: \`$\{propKey}${separator}$\{withoutSpace(value)}\` }
             }`
-            : `(key, value) => ({ className: \`$\{classNames.get(key) || hypenateProperty(key)}${separator}$\{withoutSpace(value)}\` })`
+            : `(key, value) => ({ className: \`$\{classNameByProp.get(key) || hypenateProperty(key)}${separator}$\{withoutSpace(value)}\` })`
         },
         ${utility.hasShorthand ? 'hasShorthand: true,' : ''}
         resolveShorthand: ${utility.hasShorthand ? 'resolveShorthand' : 'prop => prop'},
       }
     }
 
-    export const css = createCss(context)
+    const cssFn = createCss(context)
+    export const cssCache = new Map()
```

Comment:
We should consider using an LRU Cache here to avoid racking up too much memory. 

Creating a map cache that stores up to a certain size (say 500 entries) and resets afterward might help.